### PR TITLE
IFU-master-2021-07-02

### PR DESCRIPTION
**Pull request by rraminen**
The conflicts from the following files have been resolved:

Auto-merging setup.py
CONFLICT (content): Merge conflict in setup.py
Auto-merging op_builder/transformer.py
CONFLICT (content): Merge conflict in op_builder/transformer.py
Auto-merging op_builder/fused_lamb.py
CONFLICT (content): Merge conflict in op_builder/fused_lamb.py
Auto-merging op_builder/fused_adam.py
CONFLICT (content): Merge conflict in op_builder/fused_adam.py
Auto-merging op_builder/cpu_adam.py
CONFLICT (content): Merge conflict in op_builder/cpu_adam.py
Auto-merging op_builder/builder.py
CONFLICT (content): Merge conflict in op_builder/builder.py

[IFU-master-2021-07-02_Conflicts.zip](https://github.com/ROCmSoftwarePlatform/DeepSpeed/files/6756941/IFU-master-2021-07-02_Conflicts.zip)

CC: @jithunnair-amd

### CODE REVIEW

**Comment by okakarpa** [comment]
Can one of the admins verify this patch?

File: setup.py [hunk 1]
```diff
@@ -178,7 +189,7 @@ def create_dir_symlink(src, dest):
 cuda_version = "0.0"
 # Set hip_version to 0.0 if cpu-only
 hip_version = "0.0"
-if torch.version.cuda is not None:
+if torch_available and torch.version.cuda is not None:
     cuda_version = ".".join(torch.version.cuda.split('.')[:2])
 if torch.version.hip is not None:
```
Comment (jithunnair-amd) [inline setup.py:194]: @rraminen I think this condition also needs to consider `torch_available` based on the recent changes. Essentially, they seem to be checking for torch availability in multiple places now, and we need to make sure it's also done for any ROCm-specific checks such as these.

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (rraminen) [reply setup.py]: @jithunnair-amd, made the change.

**Review by jithunnair-amd — APPROVED** [review]
LGTM

**Pull request merged**